src/topology/separation.lean

```diff
@@ -336,6 +336,11 @@ begin
     exact this rfl },
 end
 
+lemma embedding.t2_space [topological_space β] [t2_space β] {f : α → β} (hf : embedding f) :
+  t2_space α :=
+t2_iff_nhds.mpr $ λ x y h, hf.inj $ eq_of_nhds_ne_bot
+⟨λ H, by simpa [hf.to_inducing.nhds_eq_comap, ← comap_inf, H, not_ne_bot.mpr] using h⟩
```

Comment:
After `separated_by_continuous` you can use a 1-line proof (adjusted from the proof of `subtype.t2_space`):

```
lemma embedding.t2_space [topological_space β] [t2_space β] {f : α → β} (hf : embedding f) :
  t2_space α :=
⟨λ x y h, separated_by_continuous hf.continuous (hf.inj.ne h)⟩

instance {p : α → Prop} [t2_space α] : t2_space (subtype p) :=
embedding_subtype_coe.t2_space
```